### ISSUE TITLE
Can not install the package using R devtools

### ISSUE THREAD

**Issue by smferdoushossain** [jwindle/BayesLogit#12]
Hello,

I tried to use the codes in my R by devtools. But it has not worked. I also used Rtools. Is there any Windows 64 bit issue here? My R version 3.4.4

Thanks
Ferdous

**Comment by KarinePrince** [comment]
Hello,

I am having the same issue (R version 3.4.4). see below the error messages I got.
Anyone could help on this?
I need to install a new R package that unfortunately needs BayesLogit to be first installed...

Thanks in advance!
Karine

`> install.packages("~/Downloads/BayesLogit_0.5.1.tar.gz", repos = NULL, type = "source")
* installing *source* package ‘BayesLogit’ ...
** package ‘BayesLogit’ successfully unpacked and MD5 sums checked
** libs
clang++  -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG -DUSE_R -DNDEBUG -DDISABLE_SINGLE -DNTHROW   -I/usr/local/include   -fPIC  -Wall -g -O2  -c Matrix.cpp -o Matrix.o
clang++  -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG -DUSE_R -DNDEBUG -DDISABLE_SINGLE -DNTHROW   -I/usr/local/include   -fPIC  -Wall -g -O2  -c MatrixFrame.cpp -o MatrixFrame.o
clang++  -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG -DUSE_R -DNDEBUG -DDISABLE_SINGLE -DNTHROW   -I/usr/local/include   -fPIC  -Wall -g -O2  -c RRNG.cpp -o RRNG.o
clang++  -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG -DUSE_R -DNDEBUG -DDISABLE_SINGLE -DNTHROW   -I/usr/local/include   -fPIC  -Wall -g -O2  -c RNG.cpp -o RNG.o
clang++  -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG -DUSE_R -DNDEBUG -DDISABLE_SINGLE -DNTHROW   -I/usr/local/include   -fPIC  -Wall -g -O2  -c FSF_nmix.cpp -o FSF_nmix.o
clang++  -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG -DUSE_R -DNDEBUG -DDISABLE_SINGLE -DNTHROW   -I/usr/local/include   -fPIC  -Wall -g -O2  -c LogitWrapper.cpp -o LogitWrapper.o
clang++  -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG -DUSE_R -DNDEBUG -DDISABLE_SINGLE -DNTHROW   -I/usr/local/include   -fPIC  -Wall -g -O2  -c PolyaGamma.cpp -o PolyaGamma.o
clang++  -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG -DUSE_R -DNDEBUG -DDISABLE_SINGLE -DNTHROW   -I/usr/local/include   -fPIC  -Wall -g -O2  -c PolyaGammaAlt.cpp -o PolyaGammaAlt.o
clang++  -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG -DUSE_R -DNDEBUG -DDISABLE_SINGLE -DNTHROW   -I/usr/local/include   -fPIC  -Wall -g -O2  -c PolyaGammaSP.cpp -o PolyaGammaSP.o
clang++  -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG -DUSE_R -DNDEBUG -DDISABLE_SINGLE -DNTHROW   -I/usr/local/include   -fPIC  -Wall -g -O2  -c InvertY.cpp -o InvertY.o
clang++ -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress -L/Library/Frameworks/R.framework/Resources/lib -L/usr/local/lib -o BayesLogit.so Matrix.o MatrixFrame.o RRNG.o RNG.o FSF_nmix.o LogitWrapper.o PolyaGamma.o PolyaGammaAlt.o PolyaGammaSP.o InvertY.o -L/Library/Frameworks/R.framework/Resources/lib -lRblas -L/usr/local/gfortran/lib/gcc/x86_64-apple-darwin15/6.1.0 -L/usr/local/gfortran/lib -lgfortran -lquadmath -lm -L/Library/Frameworks/R.framework/Resources/lib -lRlapack -F/Library/Frameworks/R.framework/.. -framework R -Wl,-framework -Wl,CoreFoundation
ld: warning: directory not found for option '-L/usr/local/gfortran/lib/gcc/x86_64-apple-darwin15/6.1.0'
ld: warning: directory not found for option '-L/usr/local/gfortran/lib'
ld: library not found for -lgfortran
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [BayesLogit.so] Error 1
ERROR: compilation failed for package ‘BayesLogit’
* removing ‘/Library/Frameworks/R.framework/Versions/3.4/Resources/library/BayesLogit’
Warning in install.packages :
  installation of package ‘/Users/karine_prince/Downloads/BayesLogit_0.5.1.tar.gz’ had non-zero exit status
`

**Comment by smferdoushossain** [comment]
Yes, finally I was able to do. I downloaded the BayesLogit_0.6.tar.gz zip file and installed it manually. There are some different errors I got for few days, but I manually installed all the dependent packages. I also tried with Rtools, devtools for Github, and many more dependent packages after this same error. Then what I did, I put the BayesLogit folder in the R library and also kept the similar binaries and shell files that are usually copied after package installation. Fortunately, it worked for me finally.

**Comment by KarinePrince** [comment]
Thanks for this! I finally found another solution on my side. 
I followed recommandation from a user of the new R package I was interested in, and install R Development Tools _clang_ & _gfortran_ (it seems it was coming from my Mac version or something like that).

**Comment by YunranChen** [comment]
I do the following things before follow the install instruction. It works for me. 
1.Make sure you have installed gcc (Compiler for C).
For MacOS, go to App Store, download Xcode.
Or use the commandline in terminal:
$ brew install gcc
2.Make sure you have right path for gfortran.
$ curl -O http://r.research.att.com/libs/gfortran-4.8.2-darwin13.tar.bz2
$ sudo tar fvxz gfortran-4.8.2-darwin13.tar.bz2 -C /
